consensus/src/consensus_observer/observer.rs

```diff
@@ -285,14 +163,22 @@ impl ConsensusObserver {
         debug!(LogSchema::new(LogEntry::ConsensusObserver)
             .message("Checking consensus observer progress!"));
 
-        // If we have an active subscription, verify that the subscription is still healthy
-        if self.active_observer_subscription.is_some() {
+        // Get the peer ID of the currently active subscription
+        let active_subscription_peer = self
+            .active_observer_subscription
+            .as_ref()
+            .map(|subscription| subscription.get_peer_network_id());
+
+        // If we have an active subscription, verify that the subscription
+        // is still healthy. If not, the subscription will be terminated.
+        if active_subscription_peer.is_some() {
```

Comment:
nit: this and the one below can be a match expression?

Reply:
Aah, I want them both to run, i.e., it's possible that the first statement terminates the subscription, and the second then creates a new one 😄 FWIW, I've refactored this in an upcoming PR, so hopefully it's a little cleaner. Will defer to that.